tests/src/Functional/EntityCreateAccessTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\og\Functional\EntityCreateAccessTest.
+ */
+
+namespace Drupal\Tests\og\Functional;
+
+use Drupal\og\Og;
+use Drupal\og\OgGroupAudienceHelper;
+use Drupal\simpletest\BrowserTestBase;
+use Drupal\simpletest\ContentTypeCreationTrait;
+use Drupal\simpletest\NodeCreationTrait;
+
+/**
+ * Tests access to the create entity form through the user interface.
+ *
+ * @see og_entity_create_access().
+ *
+ * @group og
+ */
+class EntityCreateAccessTest extends BrowserTestBase {
```

Comment:
Yeah, no need to change it. I was just curious :)
On Mar 23, 2016 3:45 PM, "Pieter Frenssen" <notifications@github.com> wrote:

> In tests/src/Functional/EntityCreateAccessTest.php
> <https://github.com/amitaibu/og/pull/166#discussion_r57160517>:
>
> > +namespace Drupal\Tests\og\Functional;
> > +
> > +use Drupal\og\Og;
> > +use Drupal\og\OgGroupAudienceHelper;
> > +use Drupal\simpletest\BrowserTestBase;
> > +use Drupal\simpletest\ContentTypeCreationTrait;
> > +use Drupal\simpletest\NodeCreationTrait;
> > +
> > +/**
> > + * Tests access to the create entity form through the user interface.
> > + *
> > + * @see og_entity_create_access().
> > + *
> > + * @group og
> > + */
> > +class EntityCreateAccessTest extends BrowserTestBase {
>
> Gosh, with KernelTestBase it's not really possible to test the scenario
> from the issue summary: you cannot log users in, define permissions or
> request pages. But I mainly chose to do a functional test because this is
> testing og_entity_create_access() which lives in the global namespace.
>
> Now it doesn't seem like this function is calling any other code in the
> global namespace but defers everything to static methods in classes. So it
> looks like it is possible to do this in a kerneltest too.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/amitaibu/og/pull/166/files/6721cfc3975fed2d6d5018d9e6c5e2f6cb45b085#r57160517>
>